File: core/applib/src/main/java/org/apache/isis/applib/annotation/BookmarkPolicy.java
package org.apache.isis.applib.annotation;

public enum BookmarkPolicy {
    /**
     * Can be bookmarked, and is a top-level 'root' (or parent) bookmark.
     */
    ROOT,
    /**
     * Can be bookmarked, but only as a child or some other parent/root bookmark
     */
    AS_CHILD,
    /**
     * An unimportant entity that should never be bookmarked.
     */
    NEVER
}

File: src/main/kotlin/org/ro/bs3/to/BookmarkPolicy.kt
package org.ro.bs3.to

enum class BookmarkPolicy {

    AS_ROOT,
    AS_CHILD,
    NEVER;

    fun value(): String {
        return "name()"
    }

    companion object {

        fun fromValue(v: String): BookmarkPolicy {
            return valueOf(v)
        }
    }

}
